monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/DatastoreSchemaController.java

```diff
@@ -0,0 +1,168 @@
+        final Datastore datastore = configuration.getDatastoreCatalog()
+                .getDatastore(datastoreName);
+        if (datastore == null) {
+            response.sendError(HttpServletResponse.SC_NOT_FOUND,
+                    "No such datastore: " + datastoreName);
+            return Collections.emptyMap();
+        }
+
+        String username = getUsername();
+
+        final DataContext dataContext = getDataContext(datastore);
+
+        logger.info("Serving schemas in datastore {} to user: {}.",
+                new Object[] { datastoreName, username });
+        
+        final Map<String, Object> schemas = new HashMap<>();
+        schemas.put("schemas", createSchemaList(dataContext));
+        return schemas;
+    }
+
+    private List<Map<String,Object>> createSchemaList(DataContext dataContext) {
+        List<Map<String,Object>> schemas = new ArrayList<>();
+        for (Schema schema : dataContext.getSchemas()) {
+            schemas.add(createSchemaMap(schema));
+        }
+        return schemas;
+    }
+
+    private Map<String, Object> createSchemaMap(Schema schema) {
+        final Map<String, Object> map = new HashMap<>();
+        map.put("name", schema.getName());
+        map.put("tables", createTableList(schema));
+        return map;
+    }
+
+    private List<Map<String,Object>> createTableList(Schema schema) {
+        List<Map<String,Object>> tables = new ArrayList<>();
+        for (Table table : schema.getTables()) {
+            tables.add(createTableMap(table));
+        }
+        return tables;
+    }
+
+    private Map<String, Object> createTableMap(Table table) {
+        final Map<String, Object> map = new HashMap<>();
+        map.put("name", table.getName());
+        map.put("columns", createColumnList(table));
+        return map;
+    }
+
+    private List<Map<String,Object>> createColumnList(Table table) {
+        List<Map<String,Object>> columns = new ArrayList<>();
+        for (Column column : table.getColumns()) {
+            columns.add(createColumnMap(column));
+        }
+        return columns;
+    }
+
+    private Map<String, Object> createColumnMap(Column column) {
+        final Map<String, Object> map = new HashMap<>();
+        map.put("name", column.getName());
+        map.put("number", column.getColumnNumber());
+        map.put("type", getTypeName(column));
+        map.put("size", column.getColumnSize());
+        map.put("nativeType", column.getNativeType());
+        map.put("nullable", column.isNullable());
+        map.put("remarks", column.getRemarks());
+        map.put("indexed", column.isIndexed());
+        map.put("quote", column.getQuote());
+        map.put("primaryKey", column.isPrimaryKey());
+        return map;
+    }
+
+    private String getTypeName(Column column) {
+        ColumnType type = column.getType();
+        return type == null ? null : type.getName();
+    }
+    
+    private DataContext getDataContext(Datastore datastore) {
```

Comment:
This methods closes the DatastoreConnection immediately after returning the DataContext. That means you will get into scenarios where the DataContext is not working immediately after getting it. It's not the way you're supposed to use the Datastore API :-) Here's the normal/right way:

```
try (final DatastoreConnection connection = datastore.openConnection()) {
  DataContext dataContext = connection.getDataContext();
  // do something with the DataContext here
}
```